doc/source/raysgd/v2/user_guide.rst

```diff
@@ -170,31 +258,53 @@ You can plug all of these into RaySGD with the following interface:
 
 .. code-block:: python
 
-    def train_func(config):
-        # do something
-        for x, y in dataset:
-            result = process(x)
-            ray.sgd.report(**result)
+    from ray.util.sgd import v2 as sgd
+    from ray.util.sgd.v2 import SGDCallback, Trainer
+    from typing import List, Dict
 
+    class PrintingCallback(SGDCallback):
+        def handle_result(self, results: List[Dict], **info):
+            print(results)
 
-    # TODO: Where do we pass in the logging folder?
+    def train_func():
+        for i in range(3):
+            sgd.report(epoch=i)
+
+    trainer = Trainer(backend="torch", num_workers=2)
+    trainer.start()
     result = trainer.run(
         train_func,
-        config={"worker_batch_size": 64},
-        callbacks=[sgd.MlflowCallback()]
-        dataset=dataset)
+        callbacks=[PrintingCallback()]
+    )
+    # [{'epoch': 0, '_timestamp': 1630471763, '_time_this_iter_s': 0.0020279884338378906, '_training_iteration': 1}, {'epoch': 0, '_timestamp': 1630471763, '_time_this_iter_s': 0.0014922618865966797, '_training_iteration': 1}]
+    # [{'epoch': 1, '_timestamp': 1630471763, '_time_this_iter_s': 0.0008401870727539062, '_training_iteration': 2}, {'epoch': 1, '_timestamp': 1630471763, '_time_this_iter_s': 0.0007486343383789062, '_training_iteration': 2}]
+    # [{'epoch': 2, '_timestamp': 1630471763, '_time_this_iter_s': 0.0014500617980957031, '_training_iteration': 3}, {'epoch': 2, '_timestamp': 1630471763, '_time_this_iter_s': 0.0015292167663574219, '_training_iteration': 3}]
+    trainer.shutdown()
 
-.. Here is a list of callbacks that is supported by RaySGD:
+.. Here is a list of callbacks that are supported by RaySGD:
 
+.. * JsonLoggerCallback
+.. * TBXLoggerCallback
 .. * WandbCallback
```

Comment:
These are all commented out for now! I wanted to wait until we had more than just logging callbacks before adding this back in, since the following section is `Logging Callbacks`.

Reply:
Ah sorry I missed that. Sounds good!